includes/admin/class-wc-admin-log-table-list.php

```diff
@@ -0,0 +1,246 @@
+			'critical'  => __( 'Critical',  'woocommerce' ),
+			'error'     => __( 'Error',     'woocommerce' ),
+			'warning'   => __( 'Warning',   'woocommerce' ),
+			'notice'    => __( 'Notice',    'woocommerce' ),
+			'info'      => __( 'Info',      'woocommerce' ),
+			'debug'     => __( 'Debug',     'woocommerce' ),
+		);
+
+		if ( isset( $levels[ $level_key ] ) ) {
+			return esc_html( $levels[ $level_key ] );
+		} else {
+			return '';
+		}
+	}
+
+	/**
+	 * Message column.
+	 *
+	 * @param  array $log
+	 * @return string
+	 */
+	public function column_message( $log ) {
+		return esc_html( $log['message'] );
+	}
+
+	/**
+	 * Tag column.
+	 *
+	 * @param  array $log
+	 * @return string
+	 */
+	public function column_tag( $log ) {
+		return esc_html( $log['tag'] );
+	}
+
+	/**
+	 * Get bulk actions.
+	 *
+	 * @return array
+	 */
+	protected function get_bulk_actions() {
+		return array(
+			'delete' => __( 'Delete', 'woocommerce' ),
+		);
+	}
+
+	/**
+	 * Get a list of sortable columns.
+	 *
+	 * @return array
+	 */
+	protected function get_sortable_columns() {
+		return array(
+			'timestamp' => array( 'timestamp', true ),
+			'level'     => array( 'level',     true ),
+			'tag'       => array( 'tag',       true ),
+		);
+	}
+
+	/**
+	 * Prepare table list items.
+	 */
+	public function prepare_items() {
+		global $wpdb;
+
+		$per_page = apply_filters( 'woocommerce_status_log_items_per_page', 10 );
+		$columns  = $this->get_columns();
+		$hidden   = array();
+		$sortable = $this->get_sortable_columns();
+
+		// Column headers
+		$this->_column_headers = array( $columns, $hidden, $sortable );
+
+		$current_page = $this->get_pagenum();
+		if ( 1 < $current_page ) {
+			$offset = $per_page * ( $current_page - 1 );
+		} else {
+			$offset = 0;
+		}
+
+		$level_filter = '';
+		if ( ! empty( $_REQUEST['level'] ) && WC_Log_Levels::is_valid_level( $_REQUEST['level'] ) ) {
+			$level_filter = $wpdb->prepare(
+				'AND level >= %d',
+				array( WC_Log_Levels::get_level_severity( $_REQUEST['level'] ) )
+			);
+		}
+
+		$search = '';
+		if ( ! empty( $_REQUEST['s'] ) ) {
+			$search = "AND tag LIKE '%" . esc_sql( $wpdb->esc_like( wc_clean( $_REQUEST['s'] ) ) ) . "%' ";
+		}
+
+		if ( ! empty( $_REQUEST['orderby'] ) ) {
+			switch ( $_REQUEST['orderby'] ) {
+
+				// Intentional cascade, these are valid values.
+				case 'timestamp':
+				case 'tag':
+				case 'level':
+					$order_by = $_REQUEST['orderby'];
+					break;
+
+				// Invalid $_REQUEST orderby, use default
+				default:
+					$order_by = 'log_id';
+					break;
+			}
+		} else {
+			$order_by = 'log_id';
+		}
+
+		if ( ! empty( $_REQUEST['order'] ) && 'asc' === strtolower( $_REQUEST['order'] ) ) {
+			$order_order = 'ASC';
+		} else {
+			$order_order = 'DESC';
+		}
+
+		$this->items = $wpdb->get_results(
+			"SELECT log_id, timestamp, level, message, tag
+				FROM {$wpdb->prefix}woocommerce_log
+				WHERE 1 = 1 {$level_filter} {$search}
+				ORDER BY {$order_by} {$order_order} " .
```

Comment:
I thought the whole method could be cleaner so I refactored. Have another look please.